internal/controller/scheduler/scheduler.go

```diff
@@ -96,17 +111,41 @@ func (scheduler *Scheduler) schedulingLoopIteration() error {
 			scheduler.logger.Errorf("Failed to reactively sync VMs on worker %s: %v", workerToPoke, notifyErr)
 		}
 	}
+
 	return err
 }
 
-func processVMs(vms []v1.VM) ([]v1.VM, WorkerToResources) {
+func processScheduledVM(txn storepkg.Transaction, nameToWorker map[string]v1.Worker, scheduledVM v1.VM) error {
+	worker, ok := nameToWorker[scheduledVM.Worker]
+	if !ok {
+		scheduledVM.Status = v1.VMStatusFailed
+		scheduledVM.StatusMessage = "VM is assigned to a worker that " +
+			"doesn't exist anymore"
+
+		return txn.SetVM(scheduledVM)
+	}
+
+	if time.Since(worker.LastSeen).Minutes() > 1 {
```

Comment:
Let's create an `Offline` method on the worker itself. We can also check for that in the scheduling loop and don't assign VMs to offline workers. Maybe also change 1 minute to something like 3 or even 5? I think if a worker will get offline it will stay there. And if a worker restarts then we might get a race with the worker's reporting mechanism down in this PR.

Reply:
Implemented in 0853daa.